substrate/frame/support/procedural/src/pallet/parse/call.rs

```diff
@@ -162,6 +173,31 @@ pub fn check_dispatchable_first_arg_type(ty: &syn::Type) -> syn::Result<()> {
 	Ok(())
 }
 
+/// Check the syntax is `AccountIdFor<T>`
+pub fn check_feeless_first_arg_type(ty: &syn::Type) -> syn::Result<()> {
+	pub struct CheckFeelessFirstArg;
+	impl syn::parse::Parse for CheckFeelessFirstArg {
+		fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+			input.parse::<syn::Token![&]>()?;
+			input.parse::<keyword::AccountIdFor>()?;
```

Comment:
Hm do we really need to hard-code this? Maybe someone uses `<T as super::Config>::AccountId`?  
Does it otherwise just throw a normal rust error or what happens?

Reply:
Given the suggestion to move to `OriginFor` itself, this won't be needed. Will update accordingly.